octopus-common/src/main/java/octopus/teamcity/common/createdeployment/CreateDeploymentUserData.java

```diff
@@ -0,0 +1,29 @@
+package octopus.teamcity.common.createdeployment;
+
+import java.util.List;
+import java.util.Map;
+
+import jetbrains.buildServer.util.StringUtil;
+import octopus.teamcity.common.BaseUserData;
+
+public class CreateDeploymentUserData extends BaseUserData {
+
+  private final CreateDeploymentPropertyNames KEYS = new CreateDeploymentPropertyNames();
+
+  public CreateDeploymentUserData(Map<String, String> parameters) {
+    super(parameters);
+  }
+
+  public String getProjectNameOrId() {
```

Comment:
Confirming - Project is required? I.e. this will throw if mising.

Reply:
Yep